exposee-ui/src/components/Mainpage/Main.jsx

```diff
@@ -1,29 +1,78 @@
-import "./main.css";
 import { useState, useEffect, useContext } from "react";
-import { UserContext } from "../../UserContext"; 
+import { UserContext } from "../../UserContext.js";
 import { Link } from "react-router-dom";
+import VideoPlayer from "../Videoplayer/Videoplayer.jsx";
+import "./main.css"
 
 function Main() {
-  const {user, updateUser} = useContext(UserContext);
+  const { user, updateUser } = useContext(UserContext);
+  const [videos, setVideos] = useState([]);
+  const [currentIndex, setCurrentIndex] = useState(0);
+
+  useEffect(() => {
+    const fetchVideos = async () => {
+      try {
+        const response = await fetch("http://localhost:3000/videos");
+        if (response.ok) {
+          const data = await response.json();
+          setVideos(data);
+        } else {
+          console.log("Failed to fetch videos");
+        }
+      } catch (error) {
+        console.log("Error fetching videos", error);
+      }
+    };
+
+    fetchVideos();
+  }, []);
+
+  const handleSwipeUp = () => {
+    setCurrentIndex((prevIndex) => {
+      if (prevIndex === videos.length - 1) {
+        return 0;
+      } else {
+        return prevIndex + 1;
+      }
+    });
+  };
+
   const handleLogout = () => {
     updateUser(null);
   };
 
   return (
     <div className="main">
       <header className="header">
-         <div className="user-info"> 
-           {user ? ( 
+        <div className="user-info">
+          {user ? (
             <>
               <span>Hi {user.username}! |</span>
-               <Link to= "/profile"> Go to profile Page</Link> 
+              <Link to="/profile">Go to Profile</Link>
               <button onClick={handleLogout}>Logout</button>
             </>
           ) : (
             <Link to="/login">Login</Link>
           )}
-          </div>
+        </div>
       </header>
+      <div
+        className="video-list"
+        onMouseUp={handleSwipeUp}
```

Comment:
Is the handle swipe up working already? I didn’t see it in the test plan 

I think you’re missing a flex direction horizontal to make the swipe up make more sense in the test

Reply:
it makes the cursor change to the hand or grab making the user think it they can drag  , the pan -y is what makes the vertical swipe.